### PR TITLE
activate or desactivate upnp completed

### PR DESCRIPTION
**Pull request by xkevin190**
added function to admin panel, enable and disable upnp

### CODE REVIEW

File: src/i18n/en.json [hunk 1]
```diff
@@ -138,7 +138,8 @@
     "default": "default",
     "errorSaveListen": "an error occurred while adding a new listening address",
     "addressListenMessage1": "successful changes the server listening address will be",
-    "addressListenMessage2": "successful changes the server listening address will be all available directions"
-
+    "addressListenMessage2": "successful changes the server listening address will be all available directions",
```
Comment (jeandudey) [inline src/i18n/en.json:141]:
```suggestion
    "addressListenMessage2": "successful changes the server listening address will be all available addresses",
```

File: src/utils/chatService.js [hunk 2]
```diff
@@ -158,4 +159,20 @@ export default class ChatService {
       callback(null);
     }
   }
+
+  desactivateUpnp = async () => {
```
Comment (jeandudey) [inline src/utils/chatService.js:163]:
```suggestion
  deactivateUpnp = async () => {
```

File: src/i18n/en.json [hunk 3]
```diff
@@ -138,7 +138,8 @@
...
+    "activateUpnp": "Upnp activated successfully",
+    "desactivateUpnp": "Upnp disabled"
```
Comment (jeandudey) [inline src/i18n/en.json:143]: Upnp -> UPnP

File: src/i18n/es.json [hunk 4]
```diff
@@ -134,6 +134,8 @@
     "default": "default",
     "errorSaveListen": "Ha ocurrido un error a agregar una nueva direccion de escucha",
     "addressListenMessage1": "Cambios exitosos la dirección de escucha del servidor será",
-    "addressListenMessage2": "Cambios exitosos la dirección de escucha del servidor serán todas las direcciones disponibles"
+    "addressListenMessage2": "Cambios exitosos la dirección de escucha del servidor serán todas las direcciones disponibles",
+    "activateUpnp": "Upnp activado satisfactoriamente",
+    "desactivateUpnp": "Upnp desactivado"
```
Comment (jeandudey) [inline src/i18n/es.json:139]: Same as above, Upnp -> UPnP

File: src/i18n/en.json [hunk 5]
```diff
@@ -138,7 +138,8 @@
     "default": "default",
     "errorSaveListen": "an error occurred while adding a new listening address",
     "addressListenMessage1": "successful changes the server listening address will be",
-    "addressListenMessage2": "successful changes the server listening address will be all available directions"
-
+    "addressListenMessage2": "successful changes the server listening address will be all available addresses",
+    "activateUpnp": "UPnP activated successfully",
+    "desactivateUpnp": "UPnP disabled"
   }
 }
```
Comment (luisan00) [inline src/i18n/en.json:145]: add new line here
```suggestion
}

```

File: src/i18n/es.json [hunk 6]
```diff
@@ -134,6 +134,8 @@
     "default": "default",
     "errorSaveListen": "Ha ocurrido un error a agregar una nueva direccion de escucha",
     "addressListenMessage1": "Cambios exitosos la dirección de escucha del servidor será",
-    "addressListenMessage2": "Cambios exitosos la dirección de escucha del servidor serán todas las direcciones disponibles"
+    "addressListenMessage2": "Cambios exitosos la dirección de escucha del servidor serán todas las direcciones disponibles",
+    "activateUpnp": "UPnP activado satisfactoriamente",
+    "desactivateUpnp": "UPnP desactivado"
   }
 }
```
Comment (luisan00) [inline src/i18n/es.json:141]: no new line
```suggestion
}

```

File: src/store/chats/chatAction.js [hunk 7]
```diff
@@ -373,6 +373,7 @@ export const startManualService = (callback) => async (dispatch) => {
   }
 };
 
+
```
Comment (luisan00) [inline src/store/chats/chatAction.js:376]: delete extra line 
```suggestion
```

File: src/utils/chatService.js [hunk 8]
```diff
@@ -158,4 +158,20 @@ export default class ChatService {
...
+
+  deactivateUpnp = async () => {
+    const xpriv = await bitcoin.getPrivKey();
+    await this.stop();
+    const addressListen = await AsyncStorage.getItem('AddressListen');
+    const peerID = await this.service.start(xpriv, false, addressListen);
+    this.IsActiveUpnp = false;
+    return peerID;
+  }
+
```
Comment (luisan00) [inline src/utils/chatService.js:170]: extra line
```suggestion
```

File: src/views/config/AdministrativeComponent.js [hunk 9]
```diff
@@ -12,26 +12,38 @@ import {
...
 import { connect } from 'react-redux';
-import { closeAdministrativePanel, openAdministrativePanel } from '../../store/aplication/aplicationAction';
+import AsyncStorage from '@react-native-community/async-storage';
+import { closeAdministrativePanel, openAdministrativePanel, } from '../../store/aplication/aplicationAction';
 import { startManualService, stopService, setNewDials } from '../../store/chats/chatAction';
 import { toast } from '../../utils/utils';
 import AddManualAddress from './AddManualAddress';
 import AddNewAddressListen from './AddNewAddressListen';
+import { chatService } from '../../../App';
+
```
Comment (luisan00) [inline src/views/config/AdministrativeComponent.js:22]: extra line
```suggestion
```

File: src/views/config/AdministrativeComponent.js [hunk 10]
```diff
@@ -12,26 +12,38 @@ import {
...
+      addresListen: false,
+      upnp: false
     };
   }
 
   static navigationOptions = {
     title: 'Admistrative dashboard'
   };
 
+
```
Comment (luisan00) [inline src/views/config/AdministrativeComponent.js:39]:
```suggestion
```

File: src/views/config/AdministrativeComponent.js [hunk 11]
```diff
@@ -87,6 +99,26 @@ class AdministrativeComponent extends Component {
     });
   }
 
+
```
Comment (luisan00) [inline src/views/config/AdministrativeComponent.js:102]:
```suggestion
```

**Review by luisan00 — CHANGES_REQUESTED** [review]

File: src/utils/chatService.js (hunk 2 above)
Comment (luisan00) [reply src/utils/chatService.js]: @xkevin190 if you prefer, use `enable/disable` or `enabled/disabled` for this purpose

**Comment by luisan00** [comment]
this PR also need extra work in the Unit-Test

**Comment by luisan00** [comment]
@xkevin190 some changes need to be addressed before to continue

**Review by luisan00 — APPROVED** [review]
:tada:

**Review by jeandudey — APPROVED** [review]

**Pull request merged**